server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -133,6 +134,130 @@ def validate_cdn
       end
     end
 
+    def compatible_cpu_families
+      # @deployment is provided by pre-filter run of find_deployment
```

Comment:
I would've put all of this logic in a utility class ```fusor/server/app/lib/utils/fusor``` Seems like it could be generally useful in other places if need be. And it would be easy to add a unit test around it, and makes the deployment_controller smaller.